lib/Reader/readerir.cpp

```diff
@@ -5015,14 +5264,9 @@ IRNode *GenIR::genNullCheck(IRNode *Node) {
   return Node;
 }
 
-// Generate array bounds check.
-IRNode *GenIR::genBoundsCheck(IRNode *Array, IRNode *Index) {
+void GenIR::genBoundsCheck(Value *ArrayLength, Value *Index) {
```

Comment:
genBoundsCheck doesn't do a null check anymore (doesn't call loadLen but gets the length as an ard). But I realized that I was missing a null check for md array accesses. I'm adding it to mdArrayRefAddr.